### ISSUE TITLE
No Scoring Information Available with Proteome Discoverer 2.2

### ISSUE THREAD

**Issue by ghost** [lgatto/MSnbase#269]
```r
R version 3.4.1 (2017-06-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252    LC_MONETARY=German_Germany.1252
[4] LC_NUMERIC=C                    LC_TIME=German_Germany.1252    

attached base packages:
 [1] stats4    grid      parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] party_1.2-3         strucchange_1.5-1   sandwich_2.4-0      zoo_1.8-0           modeltools_0.2-21  
 [6] mvtnorm_1.0-6       reprtree_0.6        tree_1.0-37         plotrix_3.6-6       devtools_1.13.3    
[11] bindrcpp_0.2        randomForest_4.6-12 plotly_4.7.1        ggplot2_2.2.1       shiny_1.0.5        
[16] data.table_1.10.4   dbscan_1.1-1        KPIC_2.4.0          xcms_2.99.7         MSnbase_2.3.11     
[21] ProtGenerics_1.9.1  BiocParallel_1.11.6 mzR_2.11.11         Rcpp_0.12.12        Biobase_2.37.2     
[26] BiocGenerics_0.23.1

loaded via a namespace (and not attached):
 [1] httr_1.3.1             vsn_3.45.2             tidyr_0.7.1            jsonlite_1.5          
 [5] viridisLite_0.2.0      splines_3.4.1          foreach_1.4.3          assertthat_0.2.0      
 [9] affy_1.55.0            coin_1.2-1             yaml_2.1.14            impute_1.51.0         
[13] lattice_0.20-35        glue_1.1.1             limma_3.33.10          digest_0.6.12         
[17] RColorBrewer_1.1-2     colorspace_1.3-2       htmltools_0.3.6        httpuv_1.3.5          
[21] preprocessCore_1.39.3  Matrix_1.2-10          plyr_1.8.4             MALDIquant_1.16.4     
[25] XML_3.98-1.9           pkgconfig_2.0.1        zlibbioc_1.23.0        purrr_0.2.3           
[29] xtable_1.8-2           scales_0.5.0           RANN_2.5.1             affyio_1.47.1         
[33] git2r_0.19.0           tibble_1.3.4           IRanges_2.11.17        TH.data_1.0-8         
[37] withr_2.0.0            lazyeval_0.2.0         MassSpecWavelet_1.43.0 survival_2.41-3       
[41] magrittr_1.5           mime_0.5               memoise_1.1.0          doParallel_1.0.10     
[45] MASS_7.3-47            BiocInstaller_1.27.4   tools_3.4.1            multcomp_1.4-7        
[49] S4Vectors_0.15.8       munsell_0.4.3          pcaMethods_1.69.0      compiler_3.4.1        
[53] mzID_1.15.0            rlang_0.1.2            iterators_1.0.8        htmlwidgets_0.9       
[57] crosstalk_1.0.0        GA_3.0.2               gtable_0.2.0           codetools_0.2-15      
[61] multtest_2.33.0        curl_2.8.1             R6_2.2.2               dplyr_0.7.3           
[65] bindr_0.1             
```
I have tried accessing information contained on an mzID file as instructed in the MSnbase vignette but have not been successful in my attempts. I have received the following errors:

```r
msexp <- addIdentificationData(msexp, id = identFile,verbose = FALSE)
No scoring information available
head(fData(msexp), n = 2)
spectrum acquisition.number sequence chargeState rank passThreshold experimentalMassToCharge
F1.S00001 1 52 
F1.S00002 2 6443

dSummary(msexp)
spectrumFile idFile coverage
1 Hela.mzXML 0
fData(msexp)$pepseq
NULL
head(fData(msexp), n = 70)
spectrum acquisition.number sequence chargeState rank passThreshold experimentalMassToCharge
F1.S00001 1 52 
F1.S00002 2 6443 
F1.S00003 3 6464 
F1.S00004 4 6472 
F1.S00005 5 6475
```

It seems as if the number n, in some instances is limited and cannot go beyond, say 5000. Since both the standard and recombinant protein are "pure", ions elute at least 29 mins into the run and so I expected the first few spectra to be empty but then I cannot access those spectra after 10000 which I suspect have the compounds of interest. I then tried the whole process again with an extracted Raw file that extracted only two minutes of the whole run and I still run into these problems.

```r
plot(msexp[[10]])
Error in plot_Spectrum2(x, ...) :
Please provide reporter ions if you do not want a full spectrum.
```

As far as my understanding goes, since I am carrying out label free quantification experiments, I do not need to have any reporter ions. 

The package's algorithms report that there is no scoring information available even though a quick look at the mzID file via any text reading platform shows that the file contains "Proteome Discoverer Delta Scores" which should be the scores. 

[Hela.mzid.txt](https://github.com/lgatto/MSnbase/files/1375412/Hela.mzid.txt)
[https://www.dropbox.com/sh/fuhuhw1drjuk5z0/AAALVIYBGgnDfyUSyhSRgCf7a?dl=0](url)

In the link pasted above is information on how you can retrieve the mzID, mzXML and raw files used in my analysis.

**Comment by NebuchadnezzaraHardrada** [comment]
Here is an example of the error messages I get when I used a specially extracted raw file( a time range of 1 min). 

> identFile<-"E:/Data/Kundai/RStudio/Hela/HelaExtracted2.mzID"
> msexp<-readMSData(quantfile, verbose=FALSE, centroided = TRUE)
Warning messages:
1: In min(x) : no non-missing arguments to min; returning Inf
2: In max(x) : no non-missing arguments to max; returning -Inf
> head(fData(msexp), n = 34)
         spectrum
F1.S0001        1
F1.S0002        2
F1.S0003        3
F1.S0004        4
F1.S0005        5
F1.S0006        6
F1.S0007        7
F1.S0008        8
F1.S0009        9
F1.S0010       10
F1.S0011       11
F1.S0012       12
F1.S0013       13
F1.S0014       14
F1.S0015       15

> msexp <- addIdentificationData(msexp, id = identFile,verbose = FALSE)
No scoring information available
> plot(msexp[[1568]])
Error in plot_Spectrum2(x, ...) : 
  Please provide reporter ions if you do not want a full spectrum.
> head(fData(msexp), n = 1568)

When I analyze the result file using the Proteome Discoverer viewer, I can clearly identify two peptide groups and know the scan numbers of the spectra. Using the following code:

> fData(msexp)$pepseq
NULL

I get a NULL result when the file clearly contains two identified peptide groups. I have attached a picture which shows the Consensus results tab from Proteome Discoverer.
![psms](https://user-images.githubusercontent.com/29667095/31440899-bdd7ae42-ae91-11e7-8c7d-76c9636d5824.PNG)

**Comment by lgatto** [comment]
First, one quick clarification re plotting: to plot a full spectrum and ignore any reporter ions, use `plot(msexp[[10]], full = TRUE`. This is for historical reasons and I will change this as soon as I have time, so that just `plot(msexp[[10]])` works.

Regarding the *Proteome Discoverer Delta Scores*, these are user-specified (user is software here) parameters, rather than the official controlled vocabulary. We should be able to find a ugly workaround; I'll look into it. As for the peptide sequences, I don't see them in the mzID file you shared - if I search for `DLEEAEEYK`, I only find that match in the protein sequence, but not in as a peptide sequence, hence the empty `pepseq` variable.

Bottom line is that I think Proteome Discoverer is exporting a half-baked mzID file (but that doesn't come as a surprise to me). I assume you use another search engine to actually perform the search using PD. Do you know if you could recover the mzID files from there? I will enquire on my side about that.

**Comment by NebuchadnezzaraHardrada** [comment]
![consensusworkflow](https://user-images.githubusercontent.com/29667095/31501164-c40db030-af69-11e7-8e3d-791aa74b3d94.PNG)
I just have access to a demo version of Proteome Discoverer 2.2 but a full earlier version. The demo only allows the use of two search engines as part of protein identification workflows or nodes. I have used SEQUEST in my case for both the HeLa standard and for a recombinant protein drug on both the Fusion and QExactive. Please find attached images showing the workflow I am using for both consensus and processing in PD. Obtaining result files as either mzID or mzTab is just a simple option available after result processing when one needs to export the result files, so the mzID file does not depend on the processing steps in the workflow.
![processingsteps](https://user-images.githubusercontent.com/29667095/31501156-bf5d9e42-af69-11e7-8858-389ab903a761.PNG)

**Comment by lgatto** [comment]
That's now what I meant. The file that PD exports is a somehow customised mzid, which is why things don't work (the appropriate XML tags are not present, only PS-specific ones). I should be possible to access the original search results and get the mzid exported from there. I hope to get more details tomorrow on how to do this for Mascot (we don't use SEQUEST). 

Otherwise, if you can export a spreadsheet of all identifications (including spectrum identifier), it would be possible to use that instead of mzid.

**Comment by lgatto** [comment]
Following up here - @mchiapello provided two `mzid` files for the same analysis, one produced by Mascot and PD (running Mascot). This will help understanding out a bit more what really happening here and hopefully produce a fix.
